### PR TITLE
added Workspaces feature-related endpoints and contracts

### PR DESCRIPTION
**Pull request by promoisha**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by visingla-ms** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces": {

Please add examples for all the newly added APIs.

Check ModelValidation errors in Swagger Validation Report for more details.

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:37 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>               "$ref": "./definitions.json#/definitions/WorkspaceContract"

Is PATCH method implemented in your backend such that it can update all the properties (excluding read-only properties) in WorkspaceContract model?

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:266 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/policies": {

Pagination for this API?

"x-ms-pageable": {
          "nextLinkName": "nextLink"
        },

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:343 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           },

Response 202 is not valid for Async Put requests as per ARM RPC. Please use 201 response with Azure-AsyncOperation (instead of location) for Async Put requests.

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creating-or-updating-resources-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creating-or-updating-resources-asynchronously)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:823 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "type": "string",

As this is secret value, it must be annotated with x-ms-secret: true here. And, a secret value can only be returned as Post Action response, not in any of GET/PUT/PATCH response.

[https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/definitions.json:7281 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "type": "string",

As this is secret value, it must be annotated with x-ms-secret: true here. And, a secret value can only be returned as Post Action response, not in any of GET/PUT/PATCH response.

[https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/definitions.json:7271 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "202": {

No 202 in Async Put request as explained in earlier comment.

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:1267 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>       "head": {

Why using "head" for resource check/get, why not "get"? method head should be used for fetching etag, right?

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:1507 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/notifications/{notificationName}/recipientEmails": {

Pagination?

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:1646 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>       "head": {

You mean, Get?

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:1690 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/policyFragments": {

Pagination?

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:1829 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "202": {

No 202 for Async Put requests.

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:2047 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/policyFragments/{id}/listReferences": {

Pagination?

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:2112 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "type": "string",

As this is secret value, it must be annotated with x-ms-secret: true here. And, a secret value can only be returned as Post Action response, not in any of GET/PUT/PATCH response.

[https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/definitions.json:7301 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>         "secondaryKey": {

As this is secret value, it must be annotated with x-ms-secret: true here. And, a secret value can only be returned as Post Action response, not in any of GET/PUT/PATCH response.

[https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/definitions.json:7306 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/apis/{apiId}/operations/{operationId}/policies": {

Pagination?

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:4586 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by visingla-ms** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/apis/{apiId}/policies": {

Pagination?

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:4867 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/definitions.json [hunk 1]
```diff
@@ -8346,6 +8346,316 @@
...
+          "items": {
+            "$ref": "#/definitions/WorkspaceContract"
+          },
+          "description": "Page values."
+        },
+        "count": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Total record count number across all pages."
+        },
```
Comment (visingla-ms) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/definitions.json:8364]: List API Response shouldn't have "count". It is not part of the ARM RPC (Resource Provider Contract) and will be unexpected by any generic ARM clients. You should not be adding count to any List Response.

See the public docs for ARM List Resources Response: [https://learn.microsoft.com/en-us/rest/api/resources/resources/list#resourcelistresult](https://learn.microsoft.com/en-us/rest/api/resources/resources/list#resourcelistresult).

Please apply to all such usage.

**Comment by visingla-ms** [comment]
> {

There are several Linter Errors, please address these.

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:1 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by promoisha** [comment]
@visingla-ms Almost all the resources (and all you have commented/requested a change on) are direct copies of what we already have in our spec (i.e. there's nothing new), but just put under the /workspaces resource. Considering this is a preview API version, could we proceed with the merge and we'll address the comments once we're ready to move to the stable version? (+ @solankisamir)

**Comment by solankisamir** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/apis/{apiId}/policies": {

It will always be a single resource. No need for pagination.

---
In reply to: [1489561735](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489561735) [](http://example.com/codeflow?ancestors=1489561735,1489561735)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:4867 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/apis/{apiId}/operations/{operationId}/policies": {

It will always be a single resource. No need for pagination.

---
In reply to: [1489561502](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489561502) [](http://example.com/codeflow?ancestors=1489561502,1489561502)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:4586 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/policyFragments/{id}/listReferences": {

Added a backlog for next preview 

[https://github.com/Azure/azure-rest-api-specs/issues/23418](https://github.com/Azure/azure-rest-api-specs/issues/23418)

---
In reply to: [1489559956](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489559956) [](http://example.com/codeflow?ancestors=1489559956,1489559956)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:2112 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>           "type": "string",

We only return this on POST action.

---
In reply to: [1489555591](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489555591) [](http://example.com/codeflow?ancestors=1489555591,1489555591)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/definitions.json:7281 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>           "type": "string",

We only return this on POST action.

---
In reply to: [1489555669](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489555669) [](http://example.com/codeflow?ancestors=1489555669,1489555669)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/definitions.json:7271 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>           "type": "string",

We only return this on POST action.

---
In reply to: [1489560798](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489560798) [](http://example.com/codeflow?ancestors=1489560798,1489560798)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/definitions.json:7301 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/policyFragments": {

[https://github.com/Azure/azure-rest-api-specs/issues/23418](https://github.com/Azure/azure-rest-api-specs/issues/23418)

---
In reply to: [1489559596](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489559596) [](http://example.com/codeflow?ancestors=1489559596,1489559596)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:1829 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>       "head": {

We dont have GET, only HEAD for this resource.

---
In reply to: [1489559461](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489559461) [](http://example.com/codeflow?ancestors=1489559461,1489559461)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:1690 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/notifications/{notificationName}/recipientEmails": {

[https://github.com/Azure/azure-rest-api-specs/issues/23418](https://github.com/Azure/azure-rest-api-specs/issues/23418)

---
In reply to: [1489559291](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489559291) [](http://example.com/codeflow?ancestors=1489559291,1489559291)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:1646 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/notifications/{notificationName}/recipientUsers": {

[https://github.com/Azure/azure-rest-api-specs/issues/23418](https://github.com/Azure/azure-rest-api-specs/issues/23418)

---
In reply to: [1489558392](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489558392) [](http://example.com/codeflow?ancestors=1489558392,1489558392)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:1463 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>               "$ref": "./definitions.json#/definitions/WorkspaceContract"

Yes it is same contract as PUT and all properties are updatable.

---
In reply to: [1489550195](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489550195) [](http://example.com/codeflow?ancestors=1489550195,1489550195)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:266 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/policies": {

This will only has 1 resource.

---
In reply to: [1489551346](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489551346) [](http://example.com/codeflow?ancestors=1489551346,1489551346)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:343 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>           },

[https://github.com/Azure/azure-rest-api-specs/issues/23419](https://github.com/Azure/azure-rest-api-specs/issues/23419)

---
In reply to: [1489553054](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489553054) [](http://example.com/codeflow?ancestors=1489553054,1489553054)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:823 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>           "202": {

[https://github.com/Azure/azure-rest-api-specs/issues/23419](https://github.com/Azure/azure-rest-api-specs/issues/23419)

---
In reply to: [1489558028](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489558028) [](http://example.com/codeflow?ancestors=1489558028,1489558028)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:1267 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>           "202": {

[https://github.com/Azure/azure-rest-api-specs/issues/23419](https://github.com/Azure/azure-rest-api-specs/issues/23419)

---
In reply to: [1489559834](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489559834) [](http://example.com/codeflow?ancestors=1489559834,1489559834)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:2047 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
>           "202": {

[https://github.com/Azure/azure-rest-api-specs/pull/22537/files#diff-9fa78c93f67e78617cc5760a51363474048fbe1eb01f8d11d74c33a8b5ba41b6](https://github.com/Azure/azure-rest-api-specs/pull/22537/files#diff-9fa78c93f67e78617cc5760a51363474048fbe1eb01f8d11d74c33a8b5ba41b6)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:5850 in 7fbd7c8. [](commit_id = 7fbd7c832f51a6ed07a99cb577ed2cf34b45c294, deletion_comment = False)

**Comment by rkmanda** [comment]
>       "head": {

head is valid for this use case

---
In reply to: [1489559053](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489559053) [](http://example.com/codeflow?ancestors=1489559053,1489559053)

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-09-01-preview/apimworkspaces.json:1507 in b3cf967. [](commit_id = b3cf96734fe1b31aa668f7554b6ae644721eb17f, deletion_comment = False)

**Comment by solankisamir** [comment]
@live1206 can you please approve and merge this? This is blocking our API Release.

**Comment by live1206** [comment]
> @live1206 can you please approve and merge this? This is blocking our API Release.

@solankisamir There are a bunch of [LintDiff errors](https://github.com/Azure/azure-rest-api-specs/pull/23330#issuecomment-1489199577) introduced in this PR, is it possible to fix them? If not, could you explain why?

**Comment by promoisha** [comment]
@live1206 The PR mainly introduces the new `/workspaces` parent resource with child resources, which are the same ones we already have (e.g. `/apis`, `/products`, etc). Linter errors are about those child resources.

**Review by solankisamir — APPROVED** [review]

**Review by live1206 — APPROVED** [review]

**Pull request merged**